### DISCUSSION TITLE
The "Node" tab needs to be revisited... Signals and Groups deserve a better place in the UI

### DISCUSSION THREAD

**Discussion by chexmo** [godotengine/godot-proposals#9052] [Editor]
I came from here https://github.com/godotengine/godot/issues/27629

It's a pity this didn't transcend, because it was a very interesting issue. 

Nowadays it seems like this:
![image](https://github.com/godotengine/godot-proposals/assets/30608909/fc8daab5-1fd5-4711-9485-9eb7656d5721)

The why signals and groups deserve better places in the Ui (or at least an option to have the legacy/new) are several:

  * Running the editor in big screens makes you prefer ordering stuff different
  * Although signals and groups aren't things you modify all the time, they use to be useful information to see at a glance
  * It will reduce the amount of clicks in the workflow
  * It will encourage beginners to use those functionalities
  * Maybe others, please comment
  
Proposals? The first options that come to my mind are:

  * having them both in separate dockable tabs, one for each functionality... and the user decides where to attach each
  * including an option in the Node tab to split the view and see both features at a glance (like in Filesystem tab, where you can enable to split the view and see folders in one side and files in the other)
  * make (somehow) all this information behave together in the inspector view in a way that makes sense to the user (in some way, this is very semantic as long as the signals and goups are part of the node configuration)
  
 Edit: fix redaction fault

**Comment by Calinou** [comment]
We probably have enough space to split them into separate docks now. I've never been too fond of the "subtab" UI within the Node dock either, as it's not very obvious that these two buttons effectively act like tabs (since they don't *look* like tabs).

However, if we split them into separate docks, we won't be able to offer a single tab where you can view both at once. You could always choose to use them as separate docks in separate areas, but this would mean they'd collectively take up a lot of space (the editor only has 8 dock "slots" that can be seen at a time). That said, this is only really practical if your dock is really tall – I doubt this is viable on 1920×1080 displays at 100% scale, for instance.

**Comment by Jesusemora** [comment]
In my opinion, groups should be moved to project settings and work similar to layers. They should be assigned from the inspector.

**Comment by chexmo** [comment]
> We probably have enough space to split them into separate docks now. I've never been too fond of the "subtab" UI within the Node dock either, as it's not very obvious that these two buttons effectively act like tabs (since they don't _look_ like tabs).
> 
> However, if we split them into separate docks, we won't be able to offer a single tab where you can view both at once. You could always choose to use them as separate docks in separate areas, but this would mean they'd collectively take up a lot of space (the editor only has 8 dock "slots" that can be seen at a time). That said, this is only really practical if your dock is really tall – I doubt this is viable on 1920×1080 displays at 100% scale, for instance.

I agree with much of what you said... that's why we need an open discussion about this.

The 8-slots design is both clean and limiting... we need to find something that works in combination with that line of design.

Finally, to be honest, even in my old 1366x768 notebook I'm still into proposing a new design for this.

edit: to add, in practice the 8-slot is only 4-slot if your screen isn't big enough

**Comment by chexmo** [comment]
> In my opinion, groups should be moved to project settings and work similar to layers. They should be assigned from the inspector.

Moving signals and/or groups into the inspector and maybe adding a project settings panel for their project-wide options is a very good idea in my opinion... thinking fast, the main problem with this is that it's kinda invasive/strong modification over the ux and the workflow. However, it expands it's horizons too.

That said, I think I understand why node and inspector evolved to being separated tabs. all you have in the inspector is (broadly speaking) a single resource (or at least can be treated as a resource) and the node tab stuff is kinda the stuff what doesn't fit into that resource. 

In some point the design is clashing with user experience, that's why we are debating here.

**Comment by JayMan146** [comment]
i really wish this were the case, i honestly don't see any reason for them to be sub tabs under the node tab.

**Comment by aXu-AP** [comment]
I'll throw an alternative idea here. What if we stacked them one over another? Groups are usually more limited in their size, so it would be on top.
Quick mockup:
![kuva](https://github.com/user-attachments/assets/e1b9a15e-fc99-4b31-a896-e14f0d89735d)
This way both signals and groups are viewable at a glance.

**Reply by Zireael07** [reply]
This only works on devices with pretty tall screens. Not the case with most mobile devices

**Reply by aXu-AP** [reply]
I think as long as you're not using a phone (in which case the ui has a lot of problems), there's enough space to see the groups (unless there's a ton of them) and the most important signals. When using small display, some scrolling is to be expected. Even in the current version, you need to scroll anyway to see all the signals for pretty much any node on a small screen.

**Reply by KoBeWi** [reply]
See also https://github.com/godotengine/godot-proposals/issues/9448#issuecomment-2038131505

**Reply by azur-wolve** [reply]
some of that problem solves by making both cateogries foldable
